opencensus/stats/measure_to_view_map.py

```diff
@@ -58,7 +58,7 @@ def get_view(self, view_name, timestamp):
         if view_data_list is not None:
             for view_data in view_data_list:
                 if view_data.view.name == view_name:
-                    view_data_copy = copy.deepcopy(view_data)
+                    view_data_copy = copy.copy(view_data)
```

Comment:
@songy23 I'm assuming the `deepcopy` here was defensive, so the returned `ViewData` had a separate `_tag_value_aggregation_data_map`. But this is also creating new `View`s (and associated measures and aggregations) on each call, which I don't think you want.

Changing this to `copy` means that the "copies" returned here share the aggregation data map. A better solution might be to implement `ViewData.__copy__` or come up with a new immutable view data obj for export.

Reply:
> I'm assuming the deepcopy here was defensive, so the returned ViewData had a separate _tag_value_aggregation_data_map.

That's correct.

> Changing this to copy means that the "copies" returned here share the aggregation data map. 

Ideally the `ViewData` returned should be immutable, i.e:
```python
vd1 = stats.get_view("some view")  // vd1 has data 1.0
vd1.record(1.0) // vd1 now has data 2.0
vd2 = stats.get_view("some view") // vd2 should still have data 1.0
```
So at least the `_tag_value_aggregation_data_map` should be deeply copied.

> A better solution might be to implement ViewData.__copy__ or come up with a new immutable view data obj for export.

+1 for the latter (using immutable view data obj for export). This is exactly what we do in Java.

Reply:
Thanks @songy23. I changed this to deep-copy the aggregation map only in 5ad48a3. `MeasureToViewMap` would benefit from a total rewrite.